include/materials/material_utility.h

```diff
@@ -43,10 +43,8 @@ inline double lode_angle(
     double tolerance = std::numeric_limits<double>::epsilon());
 
 //! Compute derivative of p in terms of stress sigma
-//! \param[in] stress Stress in Voigt notation where positive is tension
 //! \retval dp_dsigma Derivative of p in terms of stress sigma
-inline const Eigen::Matrix<double, 6, 1> dp_dsigma(
-    const Eigen::Matrix<double, 6, 1>& stress);
+inline const Eigen::Matrix<double, 6, 1> dp_dsigma();
```

Comment:
Why do we not have to pass the stress vector here?

Reply:
I think that's just because `dp_dsigma` returns a constant vector independent of the input stress vector.

Reply:
Agreed with @bodhinandach. `dp_dsigma` is defined as `1/3 * identity matrix` so it'll always be a constant. As seen in our code -> [here](https://github.com/cb-geo/mpm/blob/develop/include/materials/material_utility.tcc#L83-L93)

Reply:
I was double-sure this refactoring is ok when the Mohr-Coulomb and MCC tests continued to pass after refactoring without any updates to those specific test files

Reply:
@jgiven100 I think you need to refactor the rests too. I think MCC and Mohr-Coulomb call this specific function.

Reply:
@bodhinandach alread did in [this commit](https://github.com/jgiven100/mpm/commit/39012f01f81f93dcf21f4418fcfbd4906cb56b49). Otherwise I don't think all of the test cases would pass since there would be compiler errors

Reply:
Okay, I will be waiting for @ezrayst's response, otherwise, I am good!